### PR TITLE
NativeAOT/win-x86: Enable FEATURE_EH_CALLFINALLY_THUNKS

### LINKED ISSUE #99687: [NativeAot] win-x86: Funclets getting double called
Several of the runtime tests fail with a similar symptom:
```
in try
  in try
    in try
    in finally
  in filter
in filter
    in finally
  in filter
in filter
caught an exception!

FAILED!

[EXPECTED OUTPUT]
in try
  in try
    in try
    in finally
  in filter
in filter
caught an exception!

[DIFF RESULT]
< caught an exception!
---
>     in finally

```

Logs of all failed tests that hit this symptom:
[testlogs.zip](https://github.com/dotnet/runtime/files/14588250/testlogs.zip)

### PR DESCRIPTION
**Pull request by filipnavara**
Fixes #99687

### CODE REVIEW

File: src/tests/JIT/jit64/eh/basics/throwinfinallyintryfilter2.ilproj [hunk 1]
```diff
@@ -2,6 +2,7 @@
   <PropertyGroup>
     <DebugType>PdbOnly</DebugType>
     <Optimize>True</Optimize>
+    <RequiresProcessIsolation>true</RequiresProcessIsolation>
```
Comment (filipnavara) [inline src/tests/JIT/jit64/eh/basics/throwinfinallyintryfilter2.ilproj:5]:
```suggestion
```

**Comment by filipnavara** [comment]
cc @dotnet/jit-contrib

**Comment by filipnavara** [comment]
I run into some issues that may be related to this. For example, `Invariants.Tests` compiled without this change passes. With this change the stack pointer is trashed early in exception handling and causes `ret` to end up reading wrong location from stack after the catch funclet executed. I'm investigating.

**Comment by filipnavara** [comment]
Turns out the issue is unrelated to this change. It's simply lack of `EECodeManager::GetResumeSp` equivalent on the NativeAOT side (https://github.com/dotnet/runtime/pull/99688#issuecomment-2002001917). That is a separate issue that needs to be fixed.

**Comment by AndyAyersMS** [comment]
@BruceForstall perhaps you should take a look?

**Comment by filipnavara** [comment]
This is hopefully the last missing piece for the win-x86 NativeAOT support.

**Review by BruceForstall — APPROVED** [review]
Looks ok to me.

Note that it is changing exception handling behavior of the existing Linux/x86 support. We don't build or test that (that I know of), and comments here indicate Linux/x86 is already broken for other reasons, so perhaps this adds yet another thing that either improves the existing state or adds to the backlog. In any case, it makes sense that x86 with funclets and x64 with funclets are on the same plan.

**Pull request merged**